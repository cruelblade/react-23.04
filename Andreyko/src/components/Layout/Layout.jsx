import React, { Component } from 'react';
import PropTypes from "prop-types";
import Container from '@material-ui/core/Container';
import Chatlist from '../ChatList';
import Header from '../Header';
import MessageField from '../MessageField';
import './style.css';

export default class Layout extends Component {
  constructor() {
    super();
  }

  render() {
    return (
      <Container maxWidth="lg">
        <Header chatId={this.props.chatId} />
        <div className="chat-container">
          <Chatlist pickCallback={this.pickCallback} chatName={this.bots} />
          <MessageField myNickname='Стиф' chatId={this.props.chatId} />
        </div>
      </Container>
    )
  }
}

Layout.propTypes = {
  chatId: PropTypes.number,
};

Layout.defaultProps = {
  chatId: 1,
};