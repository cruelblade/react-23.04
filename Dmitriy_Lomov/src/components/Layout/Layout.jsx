import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core';
import Header from '../Header/Header';
import ChatList from '../ChatList/ChatList';

const muiStyles = theme => ({
  root: {
    display: 'flex',
    marginTop: theme.spacing(8),
  },
});

class Layout extends PureComponent {
  render() {
    const { children, classes } = this.props;

    return (
      <>
        <Header />
        <main className={classes.root}>
          <ChatList />
          {children}
        </main>
      </>
    );
  }
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default withStyles(muiStyles)(Layout);
