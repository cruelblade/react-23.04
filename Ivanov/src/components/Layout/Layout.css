.root {
    display: flex;
}

.wrapper {
    display: flex; 
    width: 100%;
     
}