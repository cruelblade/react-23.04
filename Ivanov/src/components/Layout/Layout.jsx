import React from 'react';
import PropTypes from 'prop-types'
import {CssBaseline} from '@material-ui/core'
import ChatList from '../ChatList'
import Header from '../Header'
import styles from './Layout.css'

class Layout extends React.Component {

    render() {
        const { children } = this.props;
        
        return (
            <div className={styles.root}>
                <CssBaseline />
                <Header />
                <ChatList />
                <main className={styles.wrapper}>    
                    { children }
                </main>
                
            </div>
        )
    }
}

Layout.propTypes = {
    children: PropTypes.node.isRequired,
}

export default Layout;