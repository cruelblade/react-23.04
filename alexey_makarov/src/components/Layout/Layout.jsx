import React, {Component} from "react";
import Header from "../Header/Header";
import ChatList from "../ChatList/ChatList";
import {withStyles} from "@material-ui/core/styles";

const muiStyles = theme => {
    console.log(theme);
    return {
        root: {
            display: 'flex',
            marginTop: theme.spacing(8),
        },
    };
};

class Layout extends Component{

    render() {
        const {classes, children} = this.props;
        return(
            <>
                <Header/>
                <main className={classes.root}>
                    <ChatList/>
                    {children}
                    {/*<Chat name="Ready"/>*/}
                </main>
            </>
        )
    }
}

// Layout.propTypes = {
//     children: PropTypes.node.isRequired,
// };

export default withStyles(muiStyles)(Layout);