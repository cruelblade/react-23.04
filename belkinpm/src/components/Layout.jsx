import React, { Component } from 'react';
import Header from './Header';
import ChatList from './ChatList';
import MessageField from './MessageField';

class Layout extends Component {
  state = {};

  render() {
    return (
      <div className='layout'>
        <Header />
        <div className="content">
          <ChatList />
          <MessageField />
        </div>
      </div>
    );
  }
}

export default Layout;
