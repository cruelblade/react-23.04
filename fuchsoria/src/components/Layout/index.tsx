import React from 'react';
import { Layout as AntLayout } from 'antd';
import Header from '../../components/Header';
import styles from './styles.module.scss';

const { Header: AntHeader, Footer: AntFooter } = AntLayout;

export default function Layout({ children }: { children: React.ReactNode }) {
  return (
    <AntLayout className={styles.layout}>
      <AntHeader className={styles.layoutHeader}>
        <Header />
      </AntHeader>
      <AntLayout>{children}</AntLayout>
      <AntFooter className={styles.layoutFooter}>&copy; 2020 Fuchsoria</AntFooter>
    </AntLayout>
  );
}
