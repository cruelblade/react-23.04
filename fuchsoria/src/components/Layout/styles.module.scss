.layout {
  display: block;
  margin: 0 auto;
  width: 100%;
  max-width: 1000px;
  background-color: #edeef0;

  &__header {
    padding: 0;
    background: #3e4146;
    overflow: hidden;
  }

  &__footer {
    border-top: 1px #d6d6d6 solid;
  }
}
