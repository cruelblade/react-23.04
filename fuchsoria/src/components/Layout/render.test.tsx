import React from 'react';
import { render } from '@testing-library/react';
import Layout from '../Layout';

describe('<Layout />', () => {
  it('Component should render', () => {
    const { container } = render(<Layout>Text</Layout>);

    expect(container).toMatchSnapshot();
  });
});
