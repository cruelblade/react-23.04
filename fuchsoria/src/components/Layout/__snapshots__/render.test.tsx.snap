// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Component should render 1`] = `
<div>
  <section
    class="ant-layout layout"
  >
    <header
      class="ant-layout-header layoutHeader"
    >
      <div
        class="headerContainer"
      >
        <h2
          class="headerLogo"
        >
          Messenger
        </h2>
        <ul
          class="ant-menu ant-menu-dark ant-menu-root ant-menu-horizontal"
          role="menu"
        >
          <li
            class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
            role="menuitem"
            style="display: none;"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
            >
              <span>
                ···
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
          </li>
          <li
            class="ant-menu-item ant-menu-item-only-child ant-menu-item-selected"
            role="menuitem"
          >
            Home
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
            role="menuitem"
            style="display: none;"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
            >
              <span>
                ···
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
          </li>
          <li
            class="ant-menu-item ant-menu-item-only-child"
            role="menuitem"
          >
            Chats
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
            role="menuitem"
            style="display: none;"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
            >
              <span>
                ···
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
          </li>
          <li
            class="ant-menu-item ant-menu-item-only-child"
            role="menuitem"
          >
            Profile
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
            role="menuitem"
            style="visibility: hidden; position: absolute; display: none;"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
            >
              <span>
                ···
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
          </li>
        </ul>
      </div>
    </header>
    <section
      class="ant-layout"
    >
      Text
    </section>
    <footer
      class="ant-layout-footer layoutFooter"
    >
      © 2020 Fuchsoria
    </footer>
  </section>
</div>
`;
